sys/arm/broadcom/bcm2835/bcm2835_gpio.c

```diff
@@ -289,16 +301,41 @@ bcm_gpio_set_function(struct bcm_gpio_softc *sc, uint32_t pin, uint32_t f)
 static void
 bcm_gpio_set_pud(struct bcm_gpio_softc *sc, uint32_t pin, uint32_t state)
 {
-	uint32_t bank;
-
 	/* Must be called with lock held. */
 	BCM_GPIO_LOCK_ASSERT(sc);
 
-	bank = BCM_GPIO_BANK(pin);
-	BCM_GPIO_WRITE(sc, BCM_GPIO_GPPUD(0), state);
-	BCM_GPIO_WRITE(sc, BCM_GPIO_GPPUDCLK(bank), BCM_GPIO_MASK(pin));
-	BCM_GPIO_WRITE(sc, BCM_GPIO_GPPUD(0), 0);
-	BCM_GPIO_WRITE(sc, BCM_GPIO_GPPUDCLK(bank), 0);
+	if (sc->sc_is2711 == false) { /* BCM2835 */
```

Comment:
'sc_is2711' is boolean so no need to compare it to false.

if (!sc->sc_is2711) {} is more readable.

Reply:
Or put the true case first
```
if (sc->sc_is2711) {
		u_int mask  = BCM2711_GPIO_MASK(pin);
		u_int regid = BCM2711_GPIO_REGID(pin);
		uint32_t reg;
                ...
} else { /* BCM2835 */
...
}
```